spring-cloud-starter-stream-processor-tensorflow/src/main/java/org/springframework/cloud/stream/app/tensorflow/processor/TensorflowProcessorProperties.java

```diff
@@ -46,6 +46,22 @@
 	 */
 	private int outputIndex = 0;
 
+	/**
+	 * Controls if the inference result is stored in the {@link org.springframework.messaging.Message} payload
+	 * or header.
+	 * By default (false) the result is stored in the payload. When set to true, the result will be
+	 * stored in the header using the resultHeaderName value.
+	 */
+	private boolean saveResultInHeader = false;
+
+	/**
+	 * Applicable only for `saveResultInHeader=true`, this property holds the header name used to store
+	 * the inference result. The default name is `tf_result`.
+	 * If multiple tensorflow processors are connected in a single pipeline then make sure they have
+	 * distinct resultHeaderName values to avoid overriding.
+	 */
+	private String resultHeaderName = "tf_result";
```

Comment:
@artembilan Another header related topic is about the input. If we cascade multiple TF-processors in a single pipeline and we want to preserve the original payload (e.g. one before the first TF processor) then we have to allow a TF-processor to get its input from a named-header -e.g the header used as output by the previous TF-processor.